go-controller/pkg/node/node_ip_handler_linux.go

```diff
@@ -0,0 +1,149 @@
+// +build linux
+
+package node
+
+import (
+	"net"
+	"sync"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/kube"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+
+	"github.com/vishvananda/netlink"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/klog/v2"
+	utilnet "k8s.io/utils/net"
+)
+
+type addressManager struct {
+	addresses      sets.String
+	nodeAnnotator  kube.Annotator
+	mgmtPortConfig *managementPortConfig
+	mu             sync.Mutex
+}
+
+// initializes a new address manager which will hold all the IPs on a node
+func newAddressManager(nodeAnnotator kube.Annotator, config *managementPortConfig) *addressManager {
+	mgr := &addressManager{
+		addresses:      sets.NewString(),
+		nodeAnnotator:  nodeAnnotator,
+		mgmtPortConfig: config,
+		mu:             sync.Mutex{},
+	}
+
+	klog.Info("Address manager created with IPs: %s", mgr.addresses.List())
```

Comment:
@trozet this will always be empty, so should we bother logging `addresses`?

Reply:
just leftover from a previous implementation where I did the initial listing of IPs on the host here. I'll clean it up.